dev/com.ibm.ws.cdi.1.2.weld/src/com/ibm/ws/cdi/proxy/ProxyServicesImpl.java

```diff
@@ -53,6 +53,14 @@ public void cleanup() {
 
     @Override
     public ClassLoader getClassLoader(final Class<?> proxiedBeanType) {
+
+        if (System.getSecurityManager() != null) {
+            String className = new Exception().getStackTrace()[1].getClassName();
+            if (! className.equals("org.jboss.weld.bean.proxy.ProxyFactory")) {
+                throw new RuntimeException("A potential security breach was detected");
```

Comment:
This needs a better error message. It might be better to restructure so that if `ProxyFactory` is the caller we use `doPrivileged`, and if it isn't then we don't and then we don't have to worry about throwing an exception ourselves.